ql/termstructures/yield/fittedbonddiscountcurve.hpp

```diff
@@ -197,24 +201,32 @@ namespace QuantLib {
         Real minimumCostValue() const;
         //! clone of the current object
         virtual std::auto_ptr<FittingMethod> clone() const = 0;
-		//! return whether there is a constraint at zero
-		bool constrainAtZero() const;
-		//! return weights being used
-		Array weights() const;
-		//! return optimization method being used
-		boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
-		//! open discountFunction to public
-		DiscountFactor discount(const Array& x, Time t) const;
+        //! return whether there is a constraint at zero
+        bool constrainAtZero() const;
+        //! return weights being used
+        Array weights() const;
+        //! return l2 penalties being used
+        Array l2() const;
+        //! return optimization method being used
+        boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
+        //! open discountFunction to public
+        DiscountFactor discount(const Array& x, Time t) const;
+        //! open discountFunction to public
+        Array gradients(const Array& x, Time t) const;
       protected:
-        //! constructor
+        //! constructors
         FittingMethod(bool constrainAtZero = true, const Array& weights = Array(),
                       boost::shared_ptr<OptimizationMethod> optimizationMethod
-                                          = boost::shared_ptr<OptimizationMethod>());
+                                          = boost::shared_ptr<OptimizationMethod>(),
+                      const Array& l2 = Array());
         //! rerun every time instruments/referenceDate changes
         virtual void init();
         //! discount function called by FittedBondDiscountCurve
         virtual DiscountFactor discountFunction(const Array& x,
                                                 Time t) const = 0;
+        //! gradient function called by FittedBondDiscountCurve
+        virtual Array gradientFunction(const Array& x,
+            Time t) const = 0;
```

Comment:
Adding this as a pure virtual method makes the change non backward-compatible; if someone inherited their own class from `FittingMethod`, this would make it abstract and break their compilation.  An alternative might be a default implementation that raises an exception.